lib/management/push.js

```diff
@@ -204,15 +207,23 @@ function startWSManagementClient(pushEndpoint, instanceId, token) {
         metadata.notifyBucketChange(null);
         _config.removeListener('browser-access-enabled-change',
             browserAccessChangeHandler);
-        setTimeout(startWSManagementClient, 10000, pushEndpoint,
-            instanceId, token);
+        setTimeout(startWSManagementClient, 10000, url, token);
+        connected = false;
+
+        if (cb) {
+            process.nextTick(cb);
+        }
     });
 
     ws.on('error', err => {
+        connected = false;
         logger.error('error from push server connection', {
             error: err,
             errorMessage: err.message,
         });
+        if (cb) {
```

Comment:
I think it does not matter much because the callback is going to exit the process immediately, but it might be possible to have `cb` being called twice, once in the `error` event handler and after in the `close` event which I think is always emitted when the stream does not end in a normal manner. It could be safer to add a guard for this (e.g. using `jsutil.once` from Arsenal).

Reply:
Yes once disconnected all bets are off, but better safe than sorry I'll wrap the cb with `once` for the benefit of future users.